File: ClientApp/src/app/interfaces/Product/Response/discountResponse.ts
import { BaseResponse } from "../../baseResponse";
import { ProductsDiscountResponse } from "./productsDiscountResponse";

export interface DiscountResponse extends BaseResponse {
    id: string;
    value: number;
    start: string;
    end: string;
    productsDiscounts: ProductsDiscountResponse[] | null;
}

File: ClientApp/src/app/interfaces/Product/Response/DiscountResponse.cs
namespace Contracts.Model.Product.Response;

public class DiscountResponse : BaseResponse
{
	public Guid Id { get; set; }
	public double Value { get; set; }
	public DateTimeOffset Start { get; set; } 
	public DateTimeOffset End { get; set; } 
	public ICollection<ProductsDiscountResponse>? ProductsDiscounts { get; set; }
}